pkg/webhooks/scc/scc.go

```diff
@@ -0,0 +1,207 @@
+package scc
+
+import (
+	"fmt"
+	"net/http"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName string = "scc-validation"
+	docString   string = `Managed OpenShift Customers may not modify the following default SCCs: %s`
+)
+
+var (
+	timeout int32 = 2
+	log           = logf.Log.WithName(WebhookName)
+	scope         = admissionregv1.ClusterScope
+	rules         = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{"UPDATE", "DELETE"},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{"security.openshift.io"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"securitycontextconstraints"},
+				Scope:       &scope,
+			},
+		},
+	}
+	allowedUsers = []string{
+		"system:serviceaccount:openshift-monitoring:node-exporter",
+		"system:serviceaccount:openshift-monitoring:cluster-monitoring-operator",
+	}
+	allowedGroups = []string{}
+	defaultSCCs   = []string{
+		"anyuid",
+		"hostaccess",
+		"hostmount-anyuid",
+		"hostnetwork",
+		"node-exporter",
+		"nonroot",
+		"privileged",
+		"restricted",
+		"pipelines-scc",
```

Comment:
Should the `splunkforwarder` SCC be added to this list?

Reply:
the default here covers the default SCCs created by OCP I think. the splunkforwarder was added by security team, and also it is not mentioned in the card OSD-8377.
It should not be a blocker here. And we can add it later if needed.